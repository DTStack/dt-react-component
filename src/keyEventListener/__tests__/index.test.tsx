import React from 'react';
import KeyEventListener from '../index';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
const { KeyCombiner } = KeyEventListener;
describe('test KeyCombiner suite', () => {
    const text = 'command+shift+f';
    let dom: any, element: any, asFragment: any;
    beforeEach(() => {
        dom = render(
            <KeyCombiner
                onTrigger={() => {
                    console.log('keyAction');
                }}
                keyMap={{
                    70: true,
                    91: true,
                    16: true,
                }}
            >
                <div>{text}</div>
            </KeyCombiner>
        );
        element = dom.container;
        asFragment = dom.asFragment;
    });
    afterEach(() => {
        cleanup();
    });
    test('snapshot match', () => {
        expect(asFragment()).toMatchSnapshot();
    });
    test('test key code', () => {
        fireEvent.keyPress(element, { key: 'Enter', code: 'Enter' });
    });
    test('should render the correct text', () => {
        expect(element).toBeInTheDocument();
        expect(element).toHaveTextContent(`${text}`);
    });
});
describe('test KeyEventListener suite', () => {
    const text = '任意键盘监听';
    let dom: any, element: any, asFragment: any;
    const callBack = jest.fn();
    beforeEach(() => {
        dom = render(
            <KeyEventListener
                onKeyDown={() => {
                    callBack();
                }}
            >
                {text}
            </KeyEventListener>
        );
        element = dom.container;
        asFragment = dom.asFragment;
    });
    afterEach(() => {
        cleanup();
    });
    test('snapshot match', () => {
        expect(asFragment()).toMatchSnapshot();
    });
    test('test key code', () => {
        fireEvent.keyPress(element, { key: 'Enter', code: 'Enter' });
    });
    test('should render the correct text', () => {
        expect(element).toBeInTheDocument();
        expect(element).toHaveTextContent(`${text}`);
    });
});
