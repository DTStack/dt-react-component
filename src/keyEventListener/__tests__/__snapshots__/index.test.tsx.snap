// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test KeyCombiner suite snapshot match 1`] = `
<DocumentFragment>
  <span>
    <div>
      command+shift+f
    </div>
  </span>
</DocumentFragment>
`;

exports[`test KeyEventListener suite snapshot match 1`] = `
<DocumentFragment>
  任意键盘监听
</DocumentFragment>
`;
